File: 0704-binary-search/0704-binary-search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function(nums, target) {
  let left = 0;
  let right = nums.length - 1;
  let mid = Math.ceil((left + right) / 2);

  while (mid !== left && mid !== right) {
    if (nums[mid] === target) {
      return mid;
    } else if (nums[mid] < target) {
      left = mid;
      mid = Math.ceil((left + right) / 2);
    } else if (nums[mid] > target) {
      right = mid;
      mid = Math.ceil((left + right) / 2);
    }
  }
  
  if (nums[left] === target) return left;
  if (nums[right] === target) return right;
  
  return -1;
};

File: 0704-binary-search/0704-binary-search.py
class Solution:
  def search(self, nums: List[int], target: int) -> int:
    left = 0
    right = len(nums) - 1

    while left <= right:
      mid_index = (left + right) // 2
      mid_value = nums[mid_index]

      if mid_value < target:
        left = mid_index + 1
      elif mid_value > target:
        right = mid_index - 1
      else:
        return nums.index(mid_value)
    
    return -1
